### ISSUE TITLE
Upgrading postgresql

### ISSUE THREAD

**Issue by annaswims** [department-of-veterans-affairs/va.gov-team#3706]
Postgres 12 was release in October. [We use](https://github.com/department-of-veterans-affairs/devops/blob/master/ansible/roles/postgresql/tasks/main.yml#L36) postgresql ~~9.6~~ 9.4 . 

According to https://www.postgresql.org/support/versioning/  the final release of  Postgres 9.6 will be November 11, 2021 .   But most urgently, the final release of pg 9.4 is February 13, 2020

Consider features of pg 10/11/12 that we may benefit from and write up a plan/timeline for upgrading.

**Comment by annaswims** [comment]
@tskinn says "It seems like we would need to upgrade to postgresql 11 to get the ability to export the logs to cloudwatch"

**Comment by ricetj** [comment]
need to move databases and can chat about this after that.

**Comment by annaswims** [comment]
@ricetj This is more urgent than I originally realized.  I mistakenly thought we were using 9.6, but it looks like we're using 9.4. 

`PostgreSQL 9.4 will stop receiving fixes on February 13, 2020, which is the next planned cumulative update release.`

**Comment by johnpaulashenfelter** [comment]
While we're on it, pg12 makes some breaking changes in system table structure. This affects tooling (eg admin tools, some extensions) far more than application/user data but its a pretty strong reason to consider pausing for a bit on 11.

I am *not* sure what the dependency map is with postGIS 3.0, but I'd guess part of 3.0 is related to pg12.

TL;DR let's get this prioritized.

**Comment by johnpaulashenfelter** [comment]
Checked PostGIS 3.0: This release works with PostgreSQL 9.5-12 (https://postgis.net/2019/10/20/postgis-3.0.0/)

**Comment by ricetj** [comment]
@johnpaulashenfelter @annaswims completely hear you both, we have an ATO deadline to get our DBs moved to VAEC and scanned so that is what I meant when I said we can talk after, but the move and scan need to happen by Jan 22.

**Comment by ricetj** [comment]
Hello @johnpaulashenfelter and @annaswims we are taking on these two tasks this sprint :
https://github.com/department-of-veterans-affairs/va.gov-team/issues/4993
https://github.com/department-of-veterans-affairs/va.gov-team/issues/5082
The plan here is for us to upgrade the RDS to lastest patch release and it would be great if BE tools could do the major release after these two are done. 
cc: @alexpappasoddball

**Comment by johnpaulashenfelter** [comment]
@ricetj we're talking 9.6, right? /cc @benniemosher @tskinn

**Comment by ricetj** [comment]
> @ricetj we're talking 9.6, right? /cc @benniemosher @tskinn

Yes sir

**Comment by annaswims** [comment]
Vets-api already [runs tests using pg 9.6](mdillon/postgis:9.6-alpine), so I think we're clear to upgrade to 9.6 at any point. Ops would still do the actual rds upgrade, right?

**Comment by ricetj** [comment]
> Vets-api already [runs tests using pg 9.6](mdillon/postgis:9.6-alpine), so I think we're clear to upgrade to 9.6 at any point. Ops would still do the actual rds upgrade, right?

Hey @annaswims that is what these tickets layout here: https://github.com/department-of-veterans-affairs/va.gov-team/issues/3706#issuecomment-578748495

**Comment by annaswims** [comment]
yeah, I saw that comment and was trying to clarify.  I don't know what is meant by "BE tools could do the major release".

**Comment by johnpaulashenfelter** [comment]
@ricetj we've got branches running CI on 9.6.latest, 10.latest, and 11.latest postgis, so high level of confidence we can just go straight to one of the newer versions if your team is ok with it. My understanding is its almost easier to jump up to 10 or 11.

**Comment by wyattwalter** [comment]
Looks like there's tickets for the 11 upgrade. @annaswims or @johnpaulashenfelter do we want to keep this one open for 12 or open a separate ticket for that upgrade?

**Comment by alexpappasoddball** [comment]
@wyattwalter This ticket doesn't make much sense now.  This ticket should be re-named and converted into an Epic that contains all the version updates so we can keep track of the work in an organized fashion.

**Comment by johnpaulashenfelter** [comment]
pg11 scheduled for production rollout on Th 12/13

**Comment by jbritt1** [comment]
Was postponed until Saturday 2/15/2020, but then postponed again until sometime this week (https://dsva.slack.com/archives/C52CL1PKQ/p1581613985110400).
